sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +380,487 @@ def transform(self, X):
+            # features to impute from
+            abs_corr_mat = self._get_abs_correlation_matrix(X_filled)
+
+            # Fill in each feature in the order of ordered_idx
+            for feat_idx in ordered_idx:
+                neighbor_feat_idx = self._get_neighbor_feat_idx(
+                    n_features,
+                    feat_idx,
+                    abs_corr_mat
+                )
+                X_filled, estimator = self._impute_one_feature(
+                    X_filled,
+                    mask_missing_values,
+                    feat_idx,
+                    neighbor_feat_idx
+                )
+                estimator_triplet = (
+                    feat_idx,
+                    neighbor_feat_idx,
+                    estimator
+                )
+                self._trained_estimator_triplets.append(estimator_triplet)
+
+            if iter >= self.n_burn_in:
+                results_list.append(X_filled[mask_missing_values])
+            if self.verbose:
+                print(mice_msg + 'round %d/%d, elapsed time %0.2f'
+                      % (iter + 1, total_rounds, time() - start_t))
+
+        if len(results_list) > 0:
+            X[mask_missing_values] = np.array(results_list).mean(axis=0)
+        else:
+            X[mask_missing_values] = X_filled[mask_missing_values]
+
+        return X
+
+    def transform(self, X):
+        """Imputes all missing values in X.
+
+        Parameters
+        ----------
+        X : array-like}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        X : array-like, shape (n_samples, n_features)
+            The imputed input data.
+        """
+        check_is_fitted(self, 'initial_imputer_')
+        X = check_array(X, dtype=np.float64, force_all_finite=False)
+        X = np.asarray(X, order="F")
+        mask_missing_values = _get_mask(X, self.missing_values)
+
+        # initial imputation
+        X_filled = self.initial_imputer_.transform(X)
+        X = X[:, self._valid_statistics_indexes]
+        mask_missing_values = mask_missing_values[
+                              :,
+                              self._valid_statistics_indexes
+        ]
+
+        # perform imputations
+        n_samples, n_features = X_filled.shape
+        total_rounds = self.n_burn_in + self.n_imputations
+        results_list = []
+        if total_rounds > 0:
+            total_iterations = len(self._trained_estimator_triplets)
+            imputations_per_round = total_iterations / total_rounds
+            round_index = 0
+            if self.verbose:
+                print("[MICE] Completing matrix with shape %s" % (X.shape,))
+                start_t = time()
+                mice_msg = '[MICE] Ending imputation round '
+            for i, estimator_triplet in \
+                    enumerate(self._trained_estimator_triplets):
+                feat_idx, neighbor_feat_idx, estimator = estimator_triplet
```

Comment:
I would create a private function which would take `estimator_triplet` as argument. If None, you will compute the triplet otherwise you use it. Like this you will be able to factorize fit_transform and transform easily.